// ng-twitter-timeline/esm5/lib/ng-twitter-timeline.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, ElementRef, Input } from '@angular/core';
import { NgTwitterTimelineService } from './ng-twitter-timeline.service';
var NgTwitterTimelineComponent = /** @class */ (function () {
    function NgTwitterTimelineComponent(ngTwitterTimelineService, element) {
        this.ngTwitterTimelineService = ngTwitterTimelineService;
        this.element = element;
    }
    /**
     * @return {?}
     */
    NgTwitterTimelineComponent.prototype.ngAfterViewInit = /**
     * @return {?}
     */
    function () {
        var _this = this;
        this.ngTwitterTimelineService.loadScript().subscribe((/**
         * @param {?} twttr
         * @return {?}
         */
        function (twttr) {
            /** @type {?} */
            var nativeElement = _this.element.nativeElement;
            window['twttr'].widgets.createTimeline(_this.dataSrc, nativeElement, _this.opts).then((/**
             * @param {?} embed
             * @return {?}
             */
            function success(embed) {
                console.log('Created tweet widget: ', embed);
            })).catch((/**
             * @param {?} message
             * @return {?}
             */
            function creationError(message) {
                console.log('Could not create widget: ', message);
            }));
        }), (/**
         * @param {?} err
         * @return {?}
         */
        function (err) {
            console.log('****  ERROR LOADING TWITTER WIDGET', err);
        }), (/**
         * @return {?}
         */
        function () { }));
    };
    NgTwitterTimelineComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ng-twitter-timeline',
                    template: "<div></div>"
                }] }
    ];
    /** @nocollapse */
    NgTwitterTimelineComponent.ctorParameters = function () { return [
        { type: NgTwitterTimelineService },
        { type: ElementRef }
    ]; };
    NgTwitterTimelineComponent.propDecorators = {
        dataSrc: [{ type: Input }],
        opts: [{ type: Input }]
    };
    return NgTwitterTimelineComponent;
}());
export { NgTwitterTimelineComponent };
if (false) {
    /** @type {?} */
    NgTwitterTimelineComponent.prototype.dataSrc;
    /** @type {?} */
    NgTwitterTimelineComponent.prototype.opts;
    /**
     * @type {?}
     * @private
     */
    NgTwitterTimelineComponent.prototype.ngTwitterTimelineService;
    /**
     * @type {?}
     * @private
     */
    NgTwitterTimelineComponent.prototype.element;
}
