// ng-twitter-timeline/esm2015/lib/ng-twitter-timeline.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, ElementRef, Input } from '@angular/core';
import { NgTwitterTimelineService } from './ng-twitter-timeline.service';
export class NgTwitterTimelineComponent {
    /**
     * @param {?} ngTwitterTimelineService
     * @param {?} element
     */
    constructor(ngTwitterTimelineService, element) {
        this.ngTwitterTimelineService = ngTwitterTimelineService;
        this.element = element;
    }
    /**
     * @return {?}
     */
    ngAfterViewInit() {
        this.ngTwitterTimelineService.loadScript().subscribe((/**
         * @param {?} twttr
         * @return {?}
         */
        twttr => {
            /** @type {?} */
            let nativeElement = this.element.nativeElement;
            window['twttr'].widgets.createTimeline(this.dataSrc, nativeElement, this.opts).then((/**
             * @param {?} embed
             * @return {?}
             */
            function success(embed) {
                console.log('Created tweet widget: ', embed);
            })).catch((/**
             * @param {?} message
             * @return {?}
             */
            function creationError(message) {
                console.log('Could not create widget: ', message);
            }));
        }), (/**
         * @param {?} err
         * @return {?}
         */
        err => {
            console.log('****  ERROR LOADING TWITTER WIDGET', err);
        }), (/**
         * @return {?}
         */
        () => { }));
    }
}
NgTwitterTimelineComponent.decorators = [
    { type: Component, args: [{
                selector: 'ng-twitter-timeline',
                template: `<div></div>`
            }] }
];
/** @nocollapse */
NgTwitterTimelineComponent.ctorParameters = () => [
    { type: NgTwitterTimelineService },
    { type: ElementRef }
];
NgTwitterTimelineComponent.propDecorators = {
    dataSrc: [{ type: Input }],
    opts: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    NgTwitterTimelineComponent.prototype.dataSrc;
    /** @type {?} */
    NgTwitterTimelineComponent.prototype.opts;
    /**
     * @type {?}
     * @private
     */
    NgTwitterTimelineComponent.prototype.ngTwitterTimelineService;
    /**
     * @type {?}
     * @private
     */
    NgTwitterTimelineComponent.prototype.element;
}
