// ng-twitter-timeline/fesm5/ng-twitter-timeline.js
import { Observable } from 'rxjs';
import { Injectable, NgModule, Component, ElementRef, Input, defineInjectable } from '@angular/core';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var NgTwitterTimelineService = /** @class */ (function () {
    function NgTwitterTimelineService() {
        this.TWITTER_SCRIPT_ID = 'twitter-wjs';
        this.TWITTER_WIDGET_URL = 'https://platform.twitter.com/widgets.js';
    }
    /**
     * @return {?}
     */
    NgTwitterTimelineService.prototype.loadScript = /**
     * @return {?}
     */
    function () {
        var _this = this;
        return Observable.create((/**
         * @param {?} observer
         * @return {?}
         */
        function (observer) {
            _this.startScriptLoad();
            window['twttr'].ready((/**
             * @param {?} twttr
             * @return {?}
             */
            function onLoadTwitterScript(twttr) {
                observer.next(twttr);
                observer.complete();
            }));
        }));
    };
    /**
     * @private
     * @return {?}
     */
    NgTwitterTimelineService.prototype.startScriptLoad = /**
     * @private
     * @return {?}
     */
    function () {
        window['twttr'] = ((/**
         * @param {?} d
         * @param {?} s
         * @param {?} id
         * @param {?} url
         * @return {?}
         */
        function (d, s, id, url) {
            /** @type {?} */
            var js;
            /** @type {?} */
            var fjs = d.getElementsByTagName(s)[0];
            /** @type {?} */
            var t = window['twttr'] || {};
            if (d.getElementById(id)) {
                return t;
            }
            js = d.createElement(s);
            js.id = id;
            js.src = url;
            fjs.parentNode.insertBefore(js, fjs);
            t._e = [];
            t.ready = (/**
             * @param {?} f
             * @return {?}
             */
            function (f) {
                t._e.push(f);
            });
            return t;
        })(document, 'script', this.TWITTER_SCRIPT_ID, this.TWITTER_WIDGET_URL));
    };
    NgTwitterTimelineService.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] }
    ];
    /** @nocollapse */
    NgTwitterTimelineService.ctorParameters = function () { return []; };
    /** @nocollapse */ NgTwitterTimelineService.ngInjectableDef = defineInjectable({ factory: function NgTwitterTimelineService_Factory() { return new NgTwitterTimelineService(); }, token: NgTwitterTimelineService, providedIn: "root" });
    return NgTwitterTimelineService;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var NgTwitterTimelineComponent = /** @class */ (function () {
    function NgTwitterTimelineComponent(ngTwitterTimelineService, element) {
        this.ngTwitterTimelineService = ngTwitterTimelineService;
        this.element = element;
    }
    /**
     * @return {?}
     */
    NgTwitterTimelineComponent.prototype.ngAfterViewInit = /**
     * @return {?}
     */
    function () {
        var _this = this;
        this.ngTwitterTimelineService.loadScript().subscribe((/**
         * @param {?} twttr
         * @return {?}
         */
        function (twttr) {
            /** @type {?} */
            var nativeElement = _this.element.nativeElement;
            window['twttr'].widgets.createTimeline(_this.dataSrc, nativeElement, _this.opts).then((/**
             * @param {?} embed
             * @return {?}
             */
            function success(embed) {
                console.log('Created tweet widget: ', embed);
            })).catch((/**
             * @param {?} message
             * @return {?}
             */
            function creationError(message) {
                console.log('Could not create widget: ', message);
            }));
        }), (/**
         * @param {?} err
         * @return {?}
         */
        function (err) {
            console.log('****  ERROR LOADING TWITTER WIDGET', err);
        }), (/**
         * @return {?}
         */
        function () { }));
    };
    NgTwitterTimelineComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ng-twitter-timeline',
                    template: "<div></div>"
                }] }
    ];
    /** @nocollapse */
    NgTwitterTimelineComponent.ctorParameters = function () { return [
        { type: NgTwitterTimelineService },
        { type: ElementRef }
    ]; };
    NgTwitterTimelineComponent.propDecorators = {
        dataSrc: [{ type: Input }],
        opts: [{ type: Input }]
    };
    return NgTwitterTimelineComponent;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var NgTwitterTimelineModule = /** @class */ (function () {
    function NgTwitterTimelineModule() {
    }
    NgTwitterTimelineModule.decorators = [
        { type: NgModule, args: [{
                    declarations: [NgTwitterTimelineComponent],
                    imports: [],
                    exports: [NgTwitterTimelineComponent]
                },] }
    ];
    return NgTwitterTimelineModule;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

export { NgTwitterTimelineService, NgTwitterTimelineComponent, NgTwitterTimelineModule };
