// ng-twitter-timeline/__ivy_ngcc__/fesm2015/ng-twitter-timeline.js
import { Observable } from 'rxjs';
import { Injectable, Component, ElementRef, Input, NgModule, defineInjectable } from '@angular/core';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as ɵngcc0 from '@angular/core';
class NgTwitterTimelineService {
    constructor() {
        this.TWITTER_SCRIPT_ID = 'twitter-wjs';
        this.TWITTER_WIDGET_URL = 'https://platform.twitter.com/widgets.js';
    }
    /**
     * @return {?}
     */
    loadScript() {
        return Observable.create((/**
         * @param {?} observer
         * @return {?}
         */
        observer => {
            this.startScriptLoad();
            window['twttr'].ready((/**
             * @param {?} twttr
             * @return {?}
             */
            function onLoadTwitterScript(twttr) {
                observer.next(twttr);
                observer.complete();
            }));
        }));
    }
    /**
     * @private
     * @return {?}
     */
    startScriptLoad() {
        window['twttr'] = ((/**
         * @param {?} d
         * @param {?} s
         * @param {?} id
         * @param {?} url
         * @return {?}
         */
        function (d, s, id, url) {
            /** @type {?} */
            let js;
            /** @type {?} */
            let fjs = d.getElementsByTagName(s)[0];
            /** @type {?} */
            let t = window['twttr'] || {};
            if (d.getElementById(id)) {
                return t;
            }
            js = d.createElement(s);
            js.id = id;
            js.src = url;
            fjs.parentNode.insertBefore(js, fjs);
            t._e = [];
            t.ready = (/**
             * @param {?} f
             * @return {?}
             */
            function (f) {
                t._e.push(f);
            });
            return t;
        })(document, 'script', this.TWITTER_SCRIPT_ID, this.TWITTER_WIDGET_URL));
    }
}
NgTwitterTimelineService.ɵfac = function NgTwitterTimelineService_Factory(t) { return new (t || NgTwitterTimelineService)(); };
NgTwitterTimelineService.ɵprov = /*@__PURE__*/ ɵngcc0.ɵɵdefineInjectable({ token: NgTwitterTimelineService, factory: NgTwitterTimelineService.ɵfac, providedIn: 'root' });
/** @nocollapse */
NgTwitterTimelineService.ctorParameters = () => [];
/** @nocollapse */ NgTwitterTimelineService.ngInjectableDef = defineInjectable({ factory: function NgTwitterTimelineService_Factory() { return new NgTwitterTimelineService(); }, token: NgTwitterTimelineService, providedIn: "root" });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(NgTwitterTimelineService, [{
        type: Injectable,
        args: [{
                providedIn: 'root'
            }]
    }], function () { return []; }, null); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class NgTwitterTimelineComponent {
    /**
     * @param {?} ngTwitterTimelineService
     * @param {?} element
     */
    constructor(ngTwitterTimelineService, element) {
        this.ngTwitterTimelineService = ngTwitterTimelineService;
        this.element = element;
    }
    /**
     * @return {?}
     */
    ngAfterViewInit() {
        this.ngTwitterTimelineService.loadScript().subscribe((/**
         * @param {?} twttr
         * @return {?}
         */
        twttr => {
            /** @type {?} */
            let nativeElement = this.element.nativeElement;
            window['twttr'].widgets.createTimeline(this.dataSrc, nativeElement, this.opts).then((/**
             * @param {?} embed
             * @return {?}
             */
            function success(embed) {
                console.log('Created tweet widget: ', embed);
            })).catch((/**
             * @param {?} message
             * @return {?}
             */
            function creationError(message) {
                console.log('Could not create widget: ', message);
            }));
        }), (/**
         * @param {?} err
         * @return {?}
         */
        err => {
            console.log('****  ERROR LOADING TWITTER WIDGET', err);
        }), (/**
         * @return {?}
         */
        () => { }));
    }
}
NgTwitterTimelineComponent.ɵfac = function NgTwitterTimelineComponent_Factory(t) { return new (t || NgTwitterTimelineComponent)(ɵngcc0.ɵɵdirectiveInject(NgTwitterTimelineService), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef)); };
NgTwitterTimelineComponent.ɵcmp = /*@__PURE__*/ ɵngcc0.ɵɵdefineComponent({ type: NgTwitterTimelineComponent, selectors: [["ng-twitter-timeline"]], inputs: { dataSrc: "dataSrc", opts: "opts" }, decls: 1, vars: 0, template: function NgTwitterTimelineComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelement(0, "div");
    } }, encapsulation: 2 });
/** @nocollapse */
NgTwitterTimelineComponent.ctorParameters = () => [
    { type: NgTwitterTimelineService },
    { type: ElementRef }
];
NgTwitterTimelineComponent.propDecorators = {
    dataSrc: [{ type: Input }],
    opts: [{ type: Input }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(NgTwitterTimelineComponent, [{
        type: Component,
        args: [{
                selector: 'ng-twitter-timeline',
                template: `<div></div>`
            }]
    }], function () { return [{ type: NgTwitterTimelineService }, { type: ɵngcc0.ElementRef }]; }, { dataSrc: [{
            type: Input
        }], opts: [{
            type: Input
        }] }); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class NgTwitterTimelineModule {
}
NgTwitterTimelineModule.ɵfac = function NgTwitterTimelineModule_Factory(t) { return new (t || NgTwitterTimelineModule)(); };
NgTwitterTimelineModule.ɵmod = /*@__PURE__*/ ɵngcc0.ɵɵdefineNgModule({ type: NgTwitterTimelineModule });
NgTwitterTimelineModule.ɵinj = /*@__PURE__*/ ɵngcc0.ɵɵdefineInjector({ imports: [[]] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(NgTwitterTimelineModule, [{
        type: NgModule,
        args: [{
                declarations: [NgTwitterTimelineComponent],
                imports: [],
                exports: [NgTwitterTimelineComponent]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgTwitterTimelineModule, { declarations: [NgTwitterTimelineComponent], exports: [NgTwitterTimelineComponent] }); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

export { NgTwitterTimelineService, NgTwitterTimelineComponent, NgTwitterTimelineModule };
